crates/swc_common/src/syntax_pos.rs

```diff
@@ -385,71 +385,71 @@ extern "C" {
 
 impl Span {
     #[inline]
-    pub fn lo(self) -> BytePos {
+    pub const fn lo(self) -> BytePos {
         self.lo
     }
 
     #[inline]
-    pub fn new(mut lo: BytePos, mut hi: BytePos) -> Self {
-        if lo > hi {
```

Comment:
Actually #9962 is about removing this swap

Reply:
I see. But I have two questions:
• Does this change break existing code?
• Should we add `assert!(lo < hi)` or `debug_assert!(lo < hi)`?

Reply:
Yes for the first question. (It’s a bug IMO)

And debug assert would be enough considering the number of parser tests

Reply:
Moving `swap` to the caller code will make it slower. That's why simply changing it to a `const fn` slowed it. (swap operation is inlined into call side with constant swap)

<img width="562" alt="image" src="https://github.com/user-attachments/assets/2cc532a5-748e-4be4-aa8b-8b4d6b1d60df" />